packages/backend-common/src/database/postgres.ts

```diff
@@ -77,6 +117,41 @@ export function getPgConnectionConfig(
     : connection;
 }
 
+/**
+ * Creates the missing Postgres database if it does not exist
+ *
+ * @param admin The administrative database connection, defaulting to the `postgres` database
+ * @param database The name of the database to create
+ */
+async function ensurePgDatabase(admin: knex, database: string) {
+  const result = await admin
+    .from('pg_database')
+    .where('datname', database)
+    .count<Record<string, { count: string }>>();
+
+  if (parseInt(result[0].count, 10) > 0) {
+    return;
+  }
+
+  const owner = roleCredentials(database);
+  await admin.raw(`CREATE ROLE ?? WITH LOGIN PASSWORD '${owner.password}'`, [
+    owner.user,
+  ]);
+  await admin.raw(`CREATE DATABASE ?? OWNER ??`, [database, owner.user]);
```

Comment:
What is the purpose of assigning owners to the databases? As in what problem does it solve

Reply:
This is a very simplistic approach to https://github.com/spotify/backstage/issues/1598#issuecomment-700978089 -- this very minimally prevents plugins from attempting database integration.

Reply:
Yep, good to have that prevention in place. Wondering we really need to assign passwords to the roles then, or if we can just leave that out?

Reply:
I think the hope now is that we use simple (even if predictable) passwords, in future if security requirements are such that each database gets its own credentials (via a 3rd party API -- AWS IAM, Hashicorp Vault, whatever) it should be doable. I'd prefer not to add another gotcha in future.

I agree the predictable passwords make it slightly useless (security by obscurity!), regardless :)